x-pack/plugins/security_solution/server/lib/risk_engine/risk_engine_data_client.ts

```diff
@@ -37,10 +62,58 @@ interface RiskEngineDataClientOpts {
   elasticsearchClientPromise: Promise<ElasticsearchClient>;
 }
 
+interface Configuration {
+  enable: boolean;
+  last_updated_by: string;
+}
+
 export class RiskEngineDataClient {
   private writerCache: Map<string, Writer> = new Map();
   constructor(private readonly options: RiskEngineDataClientOpts) {}
 
+  public async init({ namespace, savedObjectsClient, user }: InitOpts) {
```

Comment:
Can you elaborate please, why we need to do that and what exact logic should we move?

I understand that file is a bit big, but looks like logic is more or less related to this class.

The only things I thought made sense to move, were its exact operation with saved objects/transforms.

My concern, with extracting it to a lot of files, is that it will be harder to understand.

Reply:
In my experience, a pure function with dependencies explicitly passed in is always going to be easier to reason about than a class method leveraging a bunch of internal state/methods. 

The dependency tree for `init` is quite complicated, and this class is getting quite large. I observed that `init` receives a `savedObjectsClient`, which seems to be the center of much of this new code, and so I thought that it might be the correct place to abstract some logic into a dedicated function.

Writing this class in terms of declarative, dependency-injected functions all but guarantees those logical units can be reused later, and forces you to consider the dependencies/interaction between them by not allowing you to hide that complexity through shared state and embedded method calls. The class, then, only serves as the coordination point and central calling point of that code.

As an example: `disableLegacyTransforms` looks like:

* `disableLegacyTransforms()` : `namespace`
  * `getLegacyStatus()`: `namespace`
    * `getLegacyTransforms()`: `namespace`
  * `esClient`
  * `getLegacyTransforms()`: `namespace`
    * `esClient`

While it would be more work than leaving it as is (of course, that's always true), this could all be moved into a `risk_engine_legacy_transforms` file if `disableLegacyTransforms` and `getLegacyTransforms` are modified to accept an `esClient` instead of pulling it from an instance property.

Reply:
> The only things I thought made sense to move, were its exact operation with saved objects/transforms.

I think I agree with that, but is that not the majority of code you've added here, besides the public methods themselves? I feel like we're saying the same thing, but perhaps I've misunderstood.

Reply:
> My concern, with extracting it to a lot of files, is that it will be harder to understand.

I think the entire point of my suggestion is to identify logical groupings of code, and abstract/group them as such. Right now, the class does several things, and logic is spread amongst a multitude of private methods. At the very least, splitting that out into `risk_engine_legacy_transforms` and `risk_engine_saved_objects` will ease the mental burden of parsing through all that in a single file 👍